test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -330,8 +331,10 @@ void stopReadersAndWriters(ReaderGroupManager readerGroupManager, String readerG
         if (!FutureHelpers.await(testState.writersComplete)) {
             log.error("Writers stopped with exceptions");
         }
-        if (!FutureHelpers.await(testState.newWritersComplete)) {
-            log.error("Writers stopped with exceptions");
+        if (testState.autoScaleTestFlag.get()) {
```

Comment:
Do we need to store the flag in State object? A method `void stopWriters(boolean  includeNewWriters)`  or a better way could be used.